src/shared/fdset.c

```diff
@@ -123,6 +123,32 @@ int fdset_put_dup(FDSet *s, int fd) {
         return TAKE_FD(copy);
 }
 
+int fdset_put_indexed(FDSet *s, int fd) {
+        _cleanup_close_ int copy = -EBADF;
+        unsigned min_fd;
+        int r;
+
+        assert(s);
+        assert(fd >= 0);
+
+        /* If we store by index, then we need to ensure new FDs are always increasing monotonically,
+         * otherwise if we jump back the newly added FD won't be at the top of the ordered set anymore, as
+         * it will be of a lower value than current elements. This is used when serializing an array of FDs
+         * via SCM_RIGHTS. */
```

Comment:
i don't understand this. please explain.

are you saying SCM_RIGHTS reorders the fds? that'd be news to me.

why do have the fds to be ordered by fd nr in this set/resulting array?

Reply:
No, the set reorders, because it's an ordered set. But anyway, I've dropped the commit, so it's moot

Reply:
oh, i get it now.

Reply:
maybe say so explicitly in the comment, i.e. "FDSet is just an OrderedSet with file descriptors ordered by their numeric value (and not insertion time)".

Reply:
well, but i figure it's going to have a comeback in a later PR, no?

Reply:
yes and I've already updated the comment as suggested in the next branch